### ISSUE TITLE
VitisAI Execution Provider: Multiple Session Resource Conflict Issue

### ISSUE THREAD

**Issue by wang7393** [amd/RyzenAI-SW#223]
# VitisAI Execution Provider: Multiple Session Resource Conflict Issue

## Summary
When using VitisAI Execution Provider with multiple ONNX Runtime sessions in the same process, only the first session works correctly. Subsequent sessions cause the application to crash without any error messages or logs.

## Environment
- **Hardware**: AMD Ryzen AI NPU
- **Software**: 
  - ryzen-ai-1.5.0 conda environment
  - ONNX Runtime with VitisAI Execution Provider
  - Windows OS
- **Model**: Multiple static ONNX models with different input shapes (32, 64, 128, 160 sequence lengths)

## Expected Behavior
All VitisAI sessions should work independently without resource conflicts, allowing multiple models to be loaded and used within the same process.

## Actual Behavior
- **First session (session_32)**: Works perfectly with VitisAI
- **Subsequent sessions (session_64, session_128, session_160)**: Application crashes silently during inference
- **Memory usage**: Only the first session shows NPU memory allocation
- **No error logs**: The crash occurs without any meaningful error messages

## Reproduction Steps

1. Create multiple ONNX Runtime sessions with VitisAI provider:
```python
# This works
session_32 = onnxruntime.InferenceSession(
    model_32_path,
    providers=['VitisAIExecutionProvider', 'CPUExecutionProvider'],
    provider_options=[vitisai_options, {}]
)

# This causes crashes during inference
session_64 = onnxruntime.InferenceSession(
    model_64_path,
    providers=['VitisAIExecutionProvider', 'CPUExecutionProvider'], 
    provider_options=[vitisai_options, {}]
)
```

2. Run inference on session_32 → Works fine
3. Run inference on session_64 → Application crashes

## Detailed Log Output
```
# Session 32 loads successfully
I20250808 15:06:33.465209 27140 vitisai_compile_model.cpp:1256] AVG CPU Usage 10%
I20250808 15:06:33.465209 27140 vitisai_compile_model.cpp:1257] Peak Working Set size 749.156 MB

# Session 64 loads successfully 
I20250808 15:06:34.913609 27140 vitisai_compile_model.cpp:1256] AVG CPU Usage 0%
I20250808 15:06:34.913609 27140 vitisai_compile_model.cpp:1257] Peak Working Set size 1211.75 MB

# But crashes during inference with no error message
```

### 1. Lazy Loading Pattern
```python
# Only load one session at a time, cleanup previous session
if previous_session:
    del previous_session
    gc.collect()
    time.sleep(0.5)
# Still crashes when switching to different session
```

### 2. Deep Copy Approach
```python
# Deep copy inputs to avoid memory sharing
run_inputs = {
    "input_ids": copy.deepcopy(input_ids),
    "attention_mask": copy.deepcopy(attention_mask), 
    "decoder_input_ids": copy.deepcopy(decoder_input_ids),
}
# Still crashes
```

### 3. Hybrid Approach (Current Workaround)
```python
# Use VitisAI for primary session, CPU for others
session_32 = create_session(use_vitisai=True)   # Works
session_64 = create_session(use_vitisai=False)  # CPU fallback
session_128 = create_session(use_vitisai=False) # CPU fallback  
session_160 = create_session(use_vitisai=False) # CPU fallback
```

## Questions for Development Team

1. **Resource Management**: Does VitisAI Execution Provider support multiple active sessions in the same process?

2. **NPU Context Isolation**: Are there specific requirements for isolating NPU contexts between different model sessions?

3. **Session Lifecycle**: Should we implement a singleton pattern where only one VitisAI session is active at any given time?

4. **Configuration Options**: Are there VitisAI provider options that can help with multi-session scenarios?

## Requested Solution

We need guidance on the proper way to handle multiple ONNX models with VitisAI:

- **Option A**: Support for true multi-session VitisAI usage
- **Option B**: Official recommendation for session management patterns
- **Option C**: Clear documentation on VitisAI limitations and best practices

**Comment by wang7393** [comment]
@lakshay048 
The model is "Helsinki-NLP/opus-mt-zh-en".
And I tried exported onnx with two methold, one is a whole onnx model, and the other one is encoder + decoder model.
For the whole onnx model, I tried multi static input sequence which mentioned as above, only the first ort session can work, and the subsequent sessions cause the application to crash without any inference error messages or logs.  But from load model logs, I can see that all model load successfully, and the memory seems only include the first model's.
And also, I tried load the encoder and decoder model for a fixed input length. the decoder sessions also can not work. And the both model are loaded successfully, the memory also only include the encoder model.

For "If transformer then are you exporting ONNX model with tokenizer padding="max_length" for dynamic input sequence?", what do you mean? 
My export onnx code is:
For one whole model:
```
def export_one_static_model_160():
    """Export the translation model to ONNX format using optimum-cli."""
    # Use command-line export method (recommended approach)
    run([
        "optimum-cli", "export", "onnx", "--model", MODEL_NAME,
        "--task", "text2text-generation",
        "--opset", "17",
        "--batch_size", "1",
        "--sequence_length", "160",
        "--no-dynamic-axes",
        "--no-post-process",
        "--monolith",
        os.path.join(OUTPUT_DIR, "mmt-onnx-static-one-160")
    ], check=True)
```

For encoder + decoder model:
```
def export_static_model_128():
    """Export the translation model to ONNX format using optimum-cli."""
    # Use command-line export method (recommended approach)
    run([
        "optimum-cli", "export", "onnx", "--model", MODEL_NAME,
        "--task", "text2text-generation-with-past",
        "--opset", "17",
        "--batch_size", "1",
        "--sequence_length", "128",
        "--no-dynamic-axes",
        "--no-post-process",
        os.path.join(OUTPUT_DIR, "mmt-onnx-static-128")
    ], check=True)
```

And for tokerizer code is:
```
inputs = self.tokenizer(sentences,
                                return_tensors="np",
                                padding='max_length',
                                truncation=True,
                                max_length=self.expected_seq_length
                                )
```
